x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BooleanArrayBlock.java

```diff
@@ -50,6 +50,23 @@ private BooleanArrayBlock(
     ) {
         super(positionCount, firstValueIndexes, nulls, mvOrdering, blockFactory);
         this.vector = vector;
+        checkInvariants();
+    }
+
+    private void checkInvariants() {
```

Comment:
To make it a little more obvious this is "assertion only" code, you could make this return a dummy boolean you can `assert` it. Or you could use `if (Assertions.ENABLED == false) {return}` at the front of it.